### DISCUSSION TITLE
SB 5.1 - OCI - Build/Core/EncService docker build error

### DISCUSSION THREAD

**Discussion by subhashchandrab** [Sunbird-Ed/Community#424] [Installation]
The Jenkins build is failing with following error

```
#8 48.13 
#8 48.13 npm ERR! Linux 5.15.0-1027-oracle
#8 48.13 npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "i"
#8 48.13 npm ERR! node v6.14.4
#8 48.13 npm ERR! npm  v3.10.10
#8 48.13 npm ERR! path /home/enc/node_modules/.staging/@types/node-54e63457/package.json
#8 48.13 npm ERR! code ENOTDIR
#8 48.13 npm ERR! errno -20
#8 48.13 npm ERR! syscall open
#8 48.13 
#8 48.13 npm ERR! ENOTDIR: not a directory, open '/home/enc/node_modules/.staging/@types/node-54e63457/package.json'
#8 48.13 npm ERR! 
#8 48.13 npm ERR! If you need help, you may report this error at:
#8 48.13 npm ERR!     <https://github.com/npm/npm/issues>
#8 48.18 
#8 48.18 npm ERR! Please include the following file with any support request:
#8 48.18 npm ERR!     /home/enc/npm-debug.log
#8 ERROR: process "/bin/sh -c apt update     && apt install -y zip python make g++     && npm i     && apt remove --purge -y python make g++     && apt-get autoremove -y     && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 236
------
 > [4/4] RUN apt update     && apt install -y zip python make g++     && npm i     && apt remove --purge -y python make g++     && apt-get autoremove -y     && rm -rf /var/lib/apt/lists/*:
#8 48.13 npm ERR! errno -20
#8 48.13 npm ERR! syscall open
#8 48.13 
#8 48.13 npm ERR! ENOTDIR: not a directory, open '/home/enc/node_modules/.staging/@types/node-54e63457/package.json'
#8 48.13 npm ERR! 
#8 48.13 npm ERR! If you need help, you may report this error at:
#8 48.13 npm ERR!     <https://github.com/npm/npm/issues>
#8 48.18 
#8 48.18 npm ERR! Please include the following file with any support request:
#8 48.18 npm ERR!     /home/enc/npm-debug.log
------
Dockerfile:5
--------------------
   4 |     COPY . /home/enc
   5 | >>> RUN apt update \
   6 | >>>     && apt install -y zip python make g++ \
   7 | >>>     && npm i \
   8 | >>>     && apt remove --purge -y python make g++ \
   9 | >>>     && apt-get autoremove -y \
  10 | >>>     && rm -rf /var/lib/apt/lists/*
  11 |     EXPOSE 8013
--------------------
ERROR: failed to solve: process "/bin/sh -c apt update     && apt install -y zip python make g++     && npm i     && apt remove --purge -y python make g++     && apt-get autoremove -y     && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 236
[Pipeline] }
[Pipeline] // stage
[Pipeline] }
```

As per the thread https://github.com/Sunbird-Ed/Community/discussions/78, there is a workaround given. Even with the workaround, its not working. My requirement is to customize the code to make the appliction work with postgres database which is ssl enabled. Any suggestions will be of great help.

**Comment by subhashchandrab** [comment]
> The Jenkins build is failing with following error
> 
> ```
> #8 48.13 
> #8 48.13 npm ERR! Linux 5.15.0-1027-oracle
> #8 48.13 npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "i"
> #8 48.13 npm ERR! node v6.14.4
> #8 48.13 npm ERR! npm  v3.10.10
> #8 48.13 npm ERR! path /home/enc/node_modules/.staging/@types/node-54e63457/package.json
> #8 48.13 npm ERR! code ENOTDIR
> #8 48.13 npm ERR! errno -20
> #8 48.13 npm ERR! syscall open
> #8 48.13 
> #8 48.13 npm ERR! ENOTDIR: not a directory, open '/home/enc/node_modules/.staging/@types/node-54e63457/package.json'
> #8 48.13 npm ERR! 
> #8 48.13 npm ERR! If you need help, you may report this error at:
> #8 48.13 npm ERR!     <https://github.com/npm/npm/issues>
> #8 48.18 
> #8 48.18 npm ERR! Please include the following file with any support request:
> #8 48.18 npm ERR!     /home/enc/npm-debug.log
> #8 ERROR: process "/bin/sh -c apt update     && apt install -y zip python make g++     && npm i     && apt remove --purge -y python make g++     && apt-get autoremove -y     && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 236
> ------
>  > [4/4] RUN apt update     && apt install -y zip python make g++     && npm i     && apt remove --purge -y python make g++     && apt-get autoremove -y     && rm -rf /var/lib/apt/lists/*:
> #8 48.13 npm ERR! errno -20
> #8 48.13 npm ERR! syscall open
> #8 48.13 
> #8 48.13 npm ERR! ENOTDIR: not a directory, open '/home/enc/node_modules/.staging/@types/node-54e63457/package.json'
> #8 48.13 npm ERR! 
> #8 48.13 npm ERR! If you need help, you may report this error at:
> #8 48.13 npm ERR!     <https://github.com/npm/npm/issues>
> #8 48.18 
> #8 48.18 npm ERR! Please include the following file with any support request:
> #8 48.18 npm ERR!     /home/enc/npm-debug.log
> ------
> Dockerfile:5
> --------------------
>    4 |     COPY . /home/enc
>    5 | >>> RUN apt update \
>    6 | >>>     && apt install -y zip python make g++ \
>    7 | >>>     && npm i \
>    8 | >>>     && apt remove --purge -y python make g++ \
>    9 | >>>     && apt-get autoremove -y \
>   10 | >>>     && rm -rf /var/lib/apt/lists/*
>   11 |     EXPOSE 8013
> --------------------
> ERROR: failed to solve: process "/bin/sh -c apt update     && apt install -y zip python make g++     && npm i     && apt remove --purge -y python make g++     && apt-get autoremove -y     && rm -rf /var/lib/apt/lists/*" did not complete successfully: exit code: 236
> [Pipeline] }
> [Pipeline] // stage
> [Pipeline] }
> ```
> 
> As per the thread #78, there is a workaround given. Even with the workaround, its not working. My requirement is to customize the code to make the appliction work with postgres database which is ssl enabled. Any suggestions will be of great help.

The problem is narrowed down to the dependency on npm package sequelize": "5.21.11". Was able to confirm that the npm build goes fine if we remove this package. Tried with different versions. It still keeps giving the same error.

**Reply by chitranshu-keshav** [reply]
@gandham-santhosh can you please get back on this issue

**Comment by ddevadat** [comment]
the issue was the with node version and source list...used the below dockerfile to build the image


```
FROM node:8-buster
MAINTAINER "S M Y ALTAMASH" "smy.altamash@gmail.com"
WORKDIR /home/enc
COPY . /home/enc
RUN npm -v
RUN echo "deb http://archive.debian.org/debian stretch main" > /etc/apt/sources.list \
    && apt update \
    && apt install -y zip python make g++ \
    && npm i \
    && apt remove --purge -y python make g++ \
    && apt-get autoremove -y  \
    && rm -rf /var/lib/apt/lists/*
EXPOSE 8013
CMD sh entrypoint.sh
```
